### PR TITLE
[Docs] RGB documentation formatting corrections

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

* Add line breaks between warning notes.
* Add note on layer state data sync
* Anchor link corrections
* Join floating sentence

## Types of Changes

- [x] Documentation

## Issues Fixed or Closed by This PR

Fix format bug introduced by #15744 and #15745

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_rgb_matrix.md [hunk 1]
```diff
@@ -418,6 +418,7 @@ All RGB keycodes are currently shared with the RGBLIGHT system:
 `RGB_MODE_PLAIN`, `RGB_MODE_BREATHE`, `RGB_MODE_RAINBOW`, and `RGB_MODE_SWIRL` are the only ones that are mapped properly. The rest don't have a direct equivalent, and are not mapped.
 
 ?> `RGB_*` keycodes cannot be used with functions like `tap_code16(RGB_HUD)` as they're not USB HID keycodes. If you wish to replicate similar behaviour in custom code within your firmware (e.g. inside `encoder_update_user()` or `process_record_user()`), the equivalent [RGB functions](#functions-idfunctions) should be used instead.
+<br>
```
Comment (drashna) [inline docs/feature_rgb_matrix.md:421]:
```suggestion

```

File: docs/feature_rgblight.md [hunk 2]
```diff
@@ -76,9 +76,9 @@ Changing the **Value** sets the overall brightness.<br>
 |`RGB_MODE_RGBTEST` |`RGB_M_T` |Red, Green, Blue test animation mode                                |
 |`RGB_MODE_TWINKLE` |`RGB_M_TW`|Twinkle animation mode                                              |
 
-!> By default, if you have both the RGB Light and the [RGB Matrix](feature_rgb_matrix.md) feature enabled, these keycodes will work for both features, at the same time. You can disable the keycode functionality by defining the `*_DISABLE_KEYCODES` option for the specific feature.
-
 ?> `RGB_*` keycodes cannot be used with functions like `tap_code16(RGB_HUI)` as they're not USB HID keycodes. If you wish to replicate similar behaviour in custom code within your firmware (e.g. inside `encoder_update_user()` or `process_record_user()`), the equivalent [RGB functions](#functions-idfunctions) should be used instead.
+<br>
```
Comment (drashna) [inline docs/feature_rgblight.md:80]:
```suggestion

```

**Review by drashna — CHANGES_REQUESTED** [review]

File: docs/feature_rgblight.md (hunk 2 above)
Comment (filterpaper) [reply docs/feature_rgblight.md:80]: Without the `<br>` break, the following `! >` text box will be melded into the first when rendered on the docs page. That's the reason for this particular PR. 

What is the appropriate line break mark up for this case?

Comment (zvecr) [reply docs/feature_rgblight.md:80]:
```suggestion


```

File: docs/feature_rgb_matrix.md (hunk 1 above)
Comment (zvecr) [reply docs/feature_rgb_matrix.md:421]:
```suggestion


```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**